### PR TITLE
Recipe App PR

### PR DESCRIPTION
**Pull request by adamteddychang**
This is the final Pull Request for our Recipe App Project. 
In this Project we: 
- Created Login and Registration with Devise
- Created form to add food to the food list
- Created a Food list that displays a list of food added
- Created form to add Recipies to Recipes list
- Created Recipes list that displays the list of Recipes added
- The owner is authorized to delete their own recipe
- The recipes that are accessible to the public are displayed in a Public Recipe List 
- Clicking on Recipes leads to Recipe details 
- Recipe details display the name, cook time, prep time, and the list of ingredients. 
- Owner of the recipe can change the recipe from public to private or add new ingredients as they please. 
- Generates shopping lists from the ingredients and displays the sum of all items and the total amount of food items. 
- Made sure no N+1 queries happening
- Added navbar for easy access. 
- This project is done in collaboration between @hamid-murambiwa and @adamteddychang

### CODE REVIEW

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi team,

You have done a great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

- The PR title and description looks professional 👍
- You have shown how to define hard work through clean code 👍 


## Required Changes ♻️
_Check the comments under the review._
- Let's fix the login button, after I signed up, I couldn't login, the terminal shows unauthorized.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi team,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## To Improve 🙏

- [ ] Please fix the login functionality as it's asked by the previous reviewer. The user is not able to sign in after signing up.

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class FoodsController < ApplicationController
+  def index
+    @foods = Food.where(user_id: current_user.id)
+  end
```
Comment (julie-ify) [inline app/controllers/foods_controller.rb:4]: - [ ] Foods created by one user are not visible when any other user is logged in. Please make sure that all users can see foods created by other users but can not delete them.

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      flash[:notice] = 'Something went wrong'
+      render 'new'
+    end
+  end
+
+  def destroy
+    @recipe = Recipe.find(params[:id])
+    @recipe.destroy
+    flash[:notice] = 'The recipe was successfully destroyed.'
+    redirect_to recipes_url
```
Comment (julie-ify) [inline app/controllers/recipes_controller.rb:30]: - [ ] A user should not be able to delete ingredients added by another user. You can remove ingredient items only if you're the owner.

File: spec/views/food/food_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+RSpec.describe 'food', type: :feature do
+  describe 'index page' do
+    before(:each) do
+      visit user_session_path
+      @user = User.create!(name: '1',
+                           email: '1@first.com',
+                           password: '111111',
+                           confirmed_at: Time.now)
+
```
Comment (julie-ify) [inline spec/views/food/food_index_spec.rb:11]: - [ ] All the tests are failing. It was caused by the `confirmed_at` but remember that you removed the confirmation logic from the application so I don't think you need the `confirmed_at` here anymore. Kindly fix this.

Code_wY1wX9PLmT

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @hamid-murambiwa and @adamteddychang, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/foods_controller.rb (hunk 1 above)
Comment (DeliceLydia) [reply app/controllers/foods_controller.rb]: Well done here!

File: app/controllers/recipes_controller.rb (hunk 2 above)
Comment (DeliceLydia) [reply app/controllers/recipes_controller.rb:32]: Well addressed this! good job

File: spec/views/food/food_index_spec.rb (hunk 3 above)
Comment (DeliceLydia) [reply spec/views/food/food_index_spec.rb]: All tests are passing now! Kudos

**Review by DeliceLydia — APPROVED** [review]
## Hi @adamteddychang, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations🎉🎉

### Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**